src/components/Map.svelte

```diff
@@ -48,12 +60,47 @@
   };
 
   const removeMap = () => {
-    mapsStore.update(current => {
-      const next = current
-        .filter((m, i) => i !== map.index)
+    const getReindexedArr = (arr, index) => {
+      return arr
+        .filter((_, i) => i !== index)
         .map((item, i) => ({ ...item, index: i }));
-      return next;
-    });
```

Comment:
Can you help me understand why we need to track `index` like this (rather than depend on the array's indices)? I'm sure there's a reason for it but I'm not following at the moment.

Reply:
I ended up removing this just now. You're right it really wasn't serving a purpose. Originally I had each of these objects set up with the index and id of the map it related to, but that felt too fragile when users changed map styles so I removed the id and just didn't think to remove both.